app/controllers/admin/product_import_controller.rb

```diff
@@ -16,16 +18,60 @@ def import
     @shipping_categories = Spree::ShippingCategory.order('name ASC')
   end
 
-  def save
-    @importer = ProductImporter.new(File.new(params[:filepath]), editable_enterprises, params[:settings])
-    @importer.save_all if @importer.has_valid_entries?
+  # def save
+  #   @importer = ProductImporter.new(File.new(params[:filepath]), spree_current_user, params[:settings])
+  #   @importer.save_all if @importer.has_valid_entries?
+  #   @import_into = params[:settings][:import_into]
+  # end
+
+  def process_data
+    @importer = ProductImporter.new(File.new(params[:filepath]), spree_current_user, {start: params[:start], end: params[:end], import_into: params[:import_into]})
+
+    @importer.validate_entries
+
+    import_results = {
+      entries: @importer.entries_json,
+      reset_counts: @importer.reset_counts
+    }
+
+    render json: import_results
+  end
+
+  def save_data
+    @importer = ProductImporter.new(File.new(params[:filepath]), spree_current_user, {start: params[:start], end: params[:end], import_into: params[:import_into], settings: params[:settings]})
+
+    @importer.save_entries
+
+    save_results = {
+      results: {
+        products_created: @importer.products_created_count,
+        products_updated: @importer.products_updated_count,
+        inventory_created: @importer.inventory_created_count,
+        inventory_updated: @importer.inventory_updated_count,
+        products_reset: @importer.products_reset_count,
+      },
+      updated_ids: @importer.updated_ids,
+      errors: @importer.errors.full_messages
+    }
```

Comment:
Maybe we can move `save_results` to `ProductImporter`, like `@importer.save_results`. There is no need for the controller to know those internals IMHO